Description: Fix filter graph for openh264/libx264 encoders
 Move format conversion before pad/scale filters to ensure correct
 pixel format ordering in the filter graph pipeline.
Author: Arjen Hiemstra <ahiemstra@heimr.nl>
Origin: upstream, https://invent.kde.org/plasma/kpipewire/-/commit/d7d1f00
Bug-KDE: https://bugs.kde.org/show_bug.cgi?id=513077
--- a/src/libx264encoder.cpp
+++ b/src/libx264encoder.cpp
@@ -35,7 +35,7 @@
     // Adjust the filter graph to ensure we are using an even frame size using a
     // pad filter. Otherwise the size adjustment below will insert a row/column
     // of garbage instead of black.
-    m_filterGraphToParse = u"pad=ceil(iw/2)*2:ceil(ih/2)*2,format=pix_fmts=yuv420p"_s;
+    m_filterGraphToParse = u"format=pix_fmts=yuv420p,pad=ceil(iw/2)*2:ceil(ih/2)*2"_s;
 }

 bool LibX264Encoder::initialize(const QSize &size)
